hal/inc/ble_hal.h

```diff
@@ -860,6 +877,16 @@ int hal_ble_gap_reject_pairing(hal_ble_conn_handle_t conn_handle, void* reserved
  */
 int hal_ble_gap_set_pairing_passkey(hal_ble_conn_handle_t conn_handle, const uint8_t* passkey, void* reserved);
 
+/**
+ * Confirm the LESC numeric comparision result.
+ *
+ * @param[in]   conn_handle BLE connection handle.
+ * @param[in]   equal       True if the numeric comparison is equal, otherwise, false.
+ *
+ * @returns     0 on success, system_error_t on error.
+ */
+int hal_ble_gap_set_lesc_numeric_comparison(hal_ble_conn_handle_t conn_handle, bool equal, void* reserved);
```

Comment:
I know that you were thinking we can reject the pairing request if the numeric comparison failed, or leave the device going forward to finish the pairing process. But the fact is that user need to perform action to the numeric comparison explicitly. And without invoking this API, the SoftDevice will consider a timeout and terminate the pairing process. Besides, by invoking this API, the SoftDevice either proceed  or terminate the pairing process immediately.

While I agree that for future OOB authentication requiring OOB data, we can use a more generic API to set passkey or OOB data.

Reply:
I was more proposing something along the lines of:
```c++
int hal_ble_gap_set_auth_status(hal_ble_conn_handle_t conn_handle, hal_ble_auth_type_t type, int result, void* data, void* reserved);
```

for future extensibility (take the example above with a grain of salt, I haven't looked into OOB, so don't know how to generalize this best).

Reply:
Since we have exposed `hal_ble_gap_set_pairing_passkey()` in the dynalib in previous releases, introducing a more generic API for user set authentication info seams to be duplicated. For the OOB authentication, user need to set the OOB data. It's similar with setting a passkey, but the length is different. Do you think it's fine to deprecate the `hal_ble_gap_set_pairing_passkey()` API?

Reply:
No strong preferences honestly, but if we can generalize this, we probably should. Not the first time we are deprecating things like this and as long as the old function is kept in the dynalib and delegates to the new function it's all good.